### PR TITLE
feat: Adds statement for `\mod_book\event\chapter_viewed`.

### PR DESCRIPTION
**Pull request by garemoko**
**Description**
Adds tracking for the book chapter viewed event. (Includes tests). 

**Related Issues**
None

**PR Type**
- Enhancement

### CODE REVIEW

File: src/transformer/utils/get_activity/book_chapter.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ];
+
+    if (property_exists($chapter, 'title')) {
+        $definition['name'] = [];
+        $definition['name'][$courselang] = $chapter->title;
+    }
+
+    if (property_exists($chapter, 'content')) {
+        $definition['description'] = [];
+        $definition['description'][$courselang] = utils\get_string_html_removed($chapter->content);
```
Comment (ryasmi) [inline src/transformer/utils/get_activity/book_chapter.php]: Nice use of the util, cool that you noticed that was there 👍

Comment (ryasmi) [reply src/transformer/utils/get_activity/book_chapter.php]: How come you didn't just go with the code below?
```php
$definition['description'] = [
  $courselang => utils\get_string_html_removed($chapter->content),
];
```

File: src/transformer/utils/get_activity/book_chapter.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace src\transformer\utils\get_activity;
+defined('MOODLE_INTERNAL') || die();
+
+use src\transformer\utils as utils;
+
+function book_chapter(array $config, \stdClass $course, \stdClass $chapter) {
+    $courselang = utils\get_course_lang($course);
+    $url = $config['app_url'].'/mod/book/view.php?id=5&chapterid='.$chapter->id;
```
Comment (ryasmi) [inline src/transformer/utils/get_activity/book_chapter.php]: Did you intend to hard code the `5` in here for id param?

Comment (garemoko) [reply src/transformer/utils/get_activity/book_chapter.php]: Oops! Good spot!

Comment (ryasmi) [reply src/transformer/utils/get_activity/book_chapter.php]: No worries, I'm actually amazed I spotted that, good to know reviews actually catch some things 😆

Comment (garemoko) [reply src/transformer/utils/get_activity/book_chapter.php]: The tests are catching it too, but you saved me some head scratching to find it!

File: src/transformer/utils/get_activity/book_chapter.php (hunk 1 above)
Comment (garemoko) [reply src/transformer/utils/get_activity/book_chapter.php]: There is no good reason. I'll amend.

Comment (ryasmi) [reply src/transformer/utils/get_activity/book_chapter.php]: Cool 👍

File: src/transformer/utils/get_activity/book_chapter.php (hunk 2 above)
Comment (ryasmi) [reply src/transformer/utils/get_activity/book_chapter.php]: Ah cool no problem, hadn't actually checked the failing build yet.

**Comment by garemoko** [comment]
@ryansmith94 this is now passing tests.

**Pull request merged**